### PR TITLE
Log full output of databricks job

### PR DESCRIPTION
**Pull request by ConstantinoSchillebeeckx**
When a databricks notebook is submitted for execution, it kicks off a cascade of subflows; when this notebook fails, it's difficult to get at the logs and to find the URL of the notebook execution that was created without having to click through all those subflows. This updates the return of `jobs_runs_submit_and_wait_for_completion` with the full metadata output of the job so that it can bubble up to the parent flow that kicks off this works - this means, for example, the notebook URL can be logged at the very top.

### Example
```python
from prefect import flow, get_logger
from prefect_databricks import jobs_runs_submit_and_wait_for_completion

@flow
def my_flow():

    logger = get_logger()
    notebook_output, jobs_meta = jobs_runs_submit_and_wait_for_completion(...)

    logger.info(jobs_meta.get("run_page_url")
```

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [x] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Comment by zzstoatzz** [comment]
hi @ConstantinoSchillebeeckx - thanks for the PR!

as is, this changes the expected value from `jobs_runs_submit_and_wait_for_completion` and break all existing uses of this, which we would need to avoid.

a couple of non-breaking approaches would be possible:
- introduce a default `False` flag to this function that changes the return value from `Dict` to `Union[Dict, Tuple[Dict, Any]]`
- write a new function that returns metadata

File: src/integrations/prefect-databricks/prefect_databricks/flows.py [hunk 1]
```diff
@@ -178,10 +179,16 @@ async def jobs_runs_submit_and_wait_for_completion(
...
+            job run metadata; by default though, the method only returns notebook output
         **jobs_runs_submit_kwargs: Additional keyword arguments to pass to `jobs_runs_submit`.
 
     Returns:
-        A dictionary of task keys to its corresponding notebook output.
+        Either a dict or a tuple (depends on `return_metadata`) comprised of
+        * task_notebook_outputs: dictionary of task keys to its corresponding notebook output;
+          this is the only object returned by default from this method
+        * jobs_runs_metadata: dictionary containing IDs of the jobs runs tasks; this is only
+          return if `return_metadata=True`.
```
[the remark this answers is not in the document]
Comment (ConstantinoSchillebeeckx) [reply src/integrations/prefect-databricks/prefect_databricks/flows.py]: done https://github.com/PrefectHQ/prefect/pull/13151/commits/7af769bcd4b99a7e9252fb34b98aa5136ff35334

File: src/integrations/prefect-databricks/prefect_databricks/flows.py [hunk 2]
```diff
@@ -65,8 +65,9 @@ async def jobs_runs_submit_and_wait_for_completion(
     timeout_seconds: Optional[int] = None,
     idempotency_token: Optional[str] = None,
     access_control_list: Optional[List[AccessControlRequest]] = None,
+    return_metadata: bool = False,
     **jobs_runs_submit_kwargs: Dict[str, Any],
-) -> Dict:
+) -> Union[Dict, Tuple[Dict, Any]]:
```
[the remark this answers is not in the document]
Comment (ConstantinoSchillebeeckx) [reply src/integrations/prefect-databricks/prefect_databricks/flows.py]: done https://github.com/PrefectHQ/prefect/pull/13151/commits/7af769bcd4b99a7e9252fb34b98aa5136ff35334

**Review by zzstoatzz — APPROVED** [review]
🙌  thank you for the contribution!

**Pull request merged**